src/core/net/netif.cpp

```diff
@@ -305,6 +306,19 @@ Error Netif::SubscribeExternalMulticast(const Address &aAddress)
     {
         VerifyOrExit(cur->GetAddress() != aAddress, error = kErrorRejected);
     }
+    // Check that the address is not a solicited node multicast
+    for (const UnicastAddress *cur = mUnicastAddresses.GetHead(); cur; cur = cur->GetNext())
+    {
+        isUnicastBasedAddress = ((aAddress.GetBytes()[8] == cur->GetAddress().GetBytes()[12] &&
```

Comment:
@sarveshkumarv3, thinking more about this, I'm not sure adding such a check in OT within `otIp6SubscribeMulticastAddress()` is the best approach.

It may be better to leave filtering to the caller of this API (the platform code). This would allow the platform code invoking `otIp6SubscribeMulticastAddress()` to apply any reasonable filter mechanism. It doesn't seem quite right for OT to assume that certain addresses should be excluded.

For example, for the `posix` platform, we could add such checks in `posix/platform/netif.cpp`, where `otIp6SubscribeMulticastAddress` is called.

What are your thoughts on this?